src/captions/captions.ts

```diff
@@ -99,12 +117,25 @@ export class Captions {
       .trim();
   }
 
+  public static getPropertyType(
+    graphSchema: GraphSchemaTypeWithAccess[],
+    propertyKey: string,
+    itemType: string
+  ): PropertyType | undefined {
+    const typeGraphSchema = graphSchema.find((schemaType) => schemaType.itemType === itemType);
+    const property = (typeGraphSchema?.properties.find(
+      (property) => property.propertyKey === propertyKey
+    ) as unknown) as GraphSchemaPropertyWithAccess;
```

Comment:
It is weird that I need a cast here, but I was not able to make it work without.
Properties is an array of GraphSchemaPropertyWithAccess so the cast shouldn't be needed or at least the cast to unknown.
```
export interface GraphSchemaTypeWithAccess extends GraphSchemaType {
    access: ItemTypeAccessRightType;
    properties: Omit<GraphSchemaProperty, 'indexedAs'>[] | GraphSchemaPropertyWithAccess[];
}
```

Reply:
![image](https://user-images.githubusercontent.com/73883108/188484595-18e74ec4-b03d-4b9b-aa2d-79a59346b330.png)

![image](https://user-images.githubusercontent.com/73883108/188484643-c8910e32-15b8-44be-9937-1db28e8c1939.png)

working for me I think, could it be your IDE?

Reply:
<img width="815" alt="Screen Shot 2022-09-05 at 6 05 34 PM" src="https://user-images.githubusercontent.com/9108397/188486716-00566e23-fb22-4f5c-9e9b-3b108034ebd1.png">
this is what I get

Reply:
actually I have a weird error but only when transpiling with 'tsc': Property 'propertyType' does not exist on type 'NonNullable<NonNullable<T>>'
was that your problem?

Reply:
<img width="526" alt="Screen Shot 2022-09-05 at 6 07 38 PM" src="https://user-images.githubusercontent.com/9108397/188486964-bc9aa17f-88fc-4ed7-9be1-1ea529511454.png">

Reply:
PS: it fails also in the CI

Reply:
I think something with the Omit is not right, but not sure it is the right time to look at that. I will keep it in mind and check with backend developer after